Centaurus.Domain/Global.cs

```diff
@@ -211,5 +237,71 @@ private static async Task ApplyUpdates()
         private static Timer snapshotTimoutTimer;
 
         private static Timer snapshotRunTimer;
+
+        #region Analytics
+
+        private static async void AnalyticsManager_OnUpdate()
+        {
+            var updates = new Dictionary<BaseSubscription, SubscriptionUpdateBase>();
+            foreach (var subs in InfoConnectionManager.GetActiveSubscriptions())
+            {
+                switch (subs)
+                {
+                    case DepthsSubscription depthsSubscription:
+                        {
+                            var depth = AnalyticsManager.MarketDepthsManager.GetDepth(depthsSubscription.Market, depthsSubscription.Precision);
+
+                            if (depth != null)
+                                updates.Add(depthsSubscription, new MarketDepthUpdate { MarketDepth = depth, UpdateDate = depth.UpdatedAt });
+                        }
+                        break;
+                    case PriceHistorySubscription priceHistorySubscription:
+                        {
+                            var frames = await AnalyticsManager.OHLCManager.GetFrames(0, priceHistorySubscription.Market, priceHistorySubscription.FramePeriod);
+                            if (frames.frames.Count > 0)
+                                updates.Add(priceHistorySubscription, new PriceHistoryUpdate { Prices = frames.frames, UpdateDate = frames.frames.Max(f => f.UpdatedAt) });
+                        }
+                        break;
+                    case TradesFeedSubscription tradesFeedSubscription:
+                        {
+                            var trades = AnalyticsManager.TradesHistoryManager.GetTrades(tradesFeedSubscription.Market);
+                            updates.Add(tradesFeedSubscription, new TradesFeedUpdate { Trades = trades, UpdateDate = new DateTime(trades.FirstOrDefault()?.Timestamp ?? 0, DateTimeKind.Utc) });
+                        }
+                        break;
+                    case AllMarketTickersSubscription allMarketTickersSubscription:
+                        {
+                            var allTickers = AnalyticsManager.MarketTickersManager.GetAllTickers();
+                            if (allTickers.Count > 0)
+                                updates.Add(allMarketTickersSubscription, new AllTickersUpdate { Tickers = allTickers, UpdateDate = allTickers.Max(t => t.UpdatedAt) });
```

Comment:
You should send updates only for those markets which were updated since the last message.